lib/mongo/collection/view/readable.rb

```diff
@@ -236,6 +236,12 @@ def estimated_document_count(opts = {})
             raise ArgumentError, "Cannot call estimated_document_count when querying with a filter"
           end
 
+          %w[limit skip].each do |opt|
+            unless @options[opt].nil?
```

Comment:
Does `if @options.key?(opt}` work in this case? If yes  I think this would be a clearer statement as to the requirements.